pkg/cmd/server/bootstrappolicy/policy.go

```diff
@@ -396,7 +401,42 @@ func GetBootstrapClusterRoles() []authorizationapi.ClusterRole {
 			ObjectMeta: kapi.ObjectMeta{
 				Name: HPAControllerRoleName,
 			},
-			Rules: []authorizationapi.PolicyRule{},
+			Rules: []authorizationapi.PolicyRule{
+				// HPA Controller
+				{
+					APIGroups: []string{authorizationapi.APIGroupExtensions},
+					Verbs:     sets.NewString("get", "list"),
+					Resources: sets.NewString("horizontalpodautoscalers"),
+				},
+				{
+					APIGroups: []string{authorizationapi.APIGroupExtensions},
+					Verbs:     sets.NewString("update"),
+					Resources: sets.NewString("horizontalpodautoscalers/status"),
+				},
+				{
+					APIGroups: []string{authorizationapi.APIGroupExtensions},
+					Verbs:     sets.NewString("get", "update"),
+					Resources: sets.NewString("replicationcontrollers/scale"),
+				},
+				{
+					Verbs:     sets.NewString("get", "update"),
+					Resources: sets.NewString("deploymentconfigs/scale"),
+				},
+				{
+					Verbs:     sets.NewString("create", "update", "patch"),
+					Resources: sets.NewString("events"),
+				},
+				// Heapster MetricsClient
+				{
+					Verbs:     sets.NewString("list"),
+					Resources: sets.NewString("pods"),
+				},
+				{
+					Verbs:         sets.NewString("proxy"),
+					Resources:     sets.NewString("services/api"),
```

Comment:
https://github.com/kubernetes/kubernetes/issues/15830 is open to add `namespaces/ns/services/s/proxy`, but that doesn't exist yet